src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/services/CaseEventMessageCacheServiceTest.java

```diff
@@ -0,0 +1,129 @@
+        caseEventMessage.setSequence(1001L);
+        caseEventMessage.setCaseId("1234");
+        caseEventMessage.setMessageId("ABCD");
+        caseEventMessage.setState(MessageState.NEW);
+        caseEventMessage.setReceived(LocalDateTime.now().minusMinutes(45));
+        caseEventMessage.setEventTimestamp(LocalDateTime.now().minusMinutes(50));
+        caseEventMessage.setFromDlq(false);
+        caseEventMessage.setDeliveryCount(1);
+        caseEventMessage.setRetryCount(1);
+        caseEventMessageRepository.save(caseEventMessage);
+
+        List<CaseEventMessageEntity> messagesFromDb = IterableUtils.toList(caseEventMessageRepository.findAll());
+
+        assertThat(messagesFromDb.size()).isEqualTo(1);
+
+        List<CaseEventMessageEntity> messagesFromCache = caseEventMessageCacheService.getAllMessagesInNewState("test");
+
+        assertThat(messages).isSameAs(messagesFromCache);
+        assertNotEquals(messages, messagesFromDb);
+    }
+
+
+    @Test
+    @Sql({"classpath:sql/delete_from_case_event_messages.sql"})
+    public void should_change_after_cache_expiry_external_api() {
+        TestConfiguration.fakeTicker.advance(1, TimeUnit.HOURS);
+        List<CaseEventMessageEntity> messages = caseEventMessageCacheService.getAllMessagesInNewState("test");
+        assertThat(messages).isEmpty();
+
+        CaseEventMessageEntity caseEventMessage = new CaseEventMessageEntity();
+        caseEventMessage.setSequence(1001L);
+        caseEventMessage.setCaseId("1234");
+        caseEventMessage.setMessageId("ABCD");
+        caseEventMessage.setState(MessageState.NEW);
+        caseEventMessage.setReceived(LocalDateTime.now().minusMinutes(45));
+        caseEventMessage.setEventTimestamp(LocalDateTime.now().minusMinutes(50));
+        caseEventMessage.setFromDlq(false);
+        caseEventMessage.setDeliveryCount(1);
+        caseEventMessage.setRetryCount(1);
+        caseEventMessageRepository.save(caseEventMessage);
+        List<CaseEventMessageEntity> messagesFromDb = IterableUtils.toList(caseEventMessageRepository.findAll());
+        assertThat(messagesFromDb.size()).isEqualTo(1);
+
+        TestConfiguration.fakeTicker.advance(10, TimeUnit.MINUTES);
+
+        List<CaseEventMessageEntity> messagesFromCache = caseEventMessageCacheService.getAllMessagesInNewState("test");
+        assertThat(messagesFromCache).isEmpty();
+
+        assertThat(messages).isSameAs(messagesFromCache);
+        assertNotEquals(messages, messagesFromDb);
+
+        TestConfiguration.fakeTicker.advance(25, TimeUnit.MINUTES);
+
+        List<CaseEventMessageEntity> messagesFromCacheAgain =
+            caseEventMessageCacheService.getAllMessagesInNewState("test");
+        assertThat(messagesFromCacheAgain).isNotEmpty();
+
+        assertThat(messages).isSameAs(messagesFromCache).isNotSameAs(messagesFromCacheAgain);
+        assertEquals(messagesFromDb, messagesFromCacheAgain);
```

Comment:
Could you also use verify steps to check if it went to the cache or DB?

Reply:
Added it